src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/import.py

```diff
@@ -0,0 +1,93 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+from msrestazure.tools import is_valid_resource_id
+from azure.mgmt.containerregistry.v2018_02_01_preview.models import (
+    ImportImageParameters,
+    ImportSource,
+    ImportMode
+)
+from ._utils import (
+    validate_managed_registry,
+    get_registry_from_name
+)
+
+SOURCE_REGISTRY_MISING = "Please specify the source container registry name, login server or resource ID: "
+IMPORT_NOT_SUPPORTED = "Imports are only supported for managed registries."
+INVALID_SOURCE_IMAGE = "Please specify source image in the format '[registry.azurecr.io/]repository[:tag]' or " \
+                       "'[registry.azurecr.io/]repository@digest'"
+SOURCE_REGISTRY_NOT_FOUND = "Source registry could not be found in the current subscription. " \
+                            "Please specify the full resource ID for it: "
+NO_TTY_ERROR = "Please specify source registry ID by passing parameters to import command directly."
+REGISTRY_MISMATCH = "Registry mismatch. Please check either source-image or resource ID " \
+                    "to make sure that they are referring to the same registry and try again."
+
+
+def acr_import(cmd,
+               client,
+               registry_name,
+               source,
+               source_registry=None,
+               target_tags=None,
+               resource_group_name=None,
+               repository=None,
+               force=False):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, IMPORT_NOT_SUPPORTED)
+
+    if not source:
+        raise CLIError(INVALID_SOURCE_IMAGE)
+    source_image = source
+
+    slash = source.find('/')
+
+    if slash < 0:
+        if not source_registry:
+            from knack.prompting import prompt, NoTTYException
+            try:
+                source_registry = prompt(SOURCE_REGISTRY_MISING)
+            except NoTTYException:
+                raise CLIError(NO_TTY_ERROR)
+        registry_from_source = get_registry_from_name(cmd.cli_ctx, source_registry, source_registry)
```

Comment:
If `source_registry` is already a resource ID, this request would be a waste since it must return None, or you need to support get registry from resource ID.